controllers/src/main/java/io/syndesis/controllers/integration/IntegrationController.java

```diff
@@ -197,10 +197,17 @@ private void callStatusChangeHandler(StatusChangeHandlerProvider.StatusChangeHan
                         .withCurrentState(IntegrationRevisionState.from(update.getStatus()));
 
                     //replace revision
-                    revisions.remove(revision);
+                    integration.getDeployedRevision().map(revisions::remove);
```

Comment:
We already have a reference of `revision` and `revsisions` what is the purpose of re-obtaining the revision (given that we only compare for equality the revisionId and ignore the rest of the object)?